WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductSelectorViewModelTests.swift

```diff
@@ -365,6 +386,26 @@ final class ProductSelectorViewModelTests: XCTestCase {
         XCTAssertEqual(productRow?.selectedState, .notSelected)
     }
 
+    func test_selecting_a_product_when_onProductSelected_is_invoked_then_sets_its_row_to_notSelected_state_if_it_was_previously_selected() {
+        // Given
+        let product = Product.fake().copy(siteID: sampleSiteID, productID: 1, purchasable: true)
+        insert(product)
+        let viewModel = ProductSelectorViewModel(siteID: sampleSiteID,
+                                                 storageManager: storageManager,
+                                                 onProductSelected: { _ in })
+
+        // When
+        viewModel.selectProduct(product.productID)
+        viewModel.selectProduct(product.productID)
+
+        // Then
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productMultiSelectionM1) {
```

Comment:
Same comment as above about removing the check with feature flag service.

Reply:
Changed: https://github.com/woocommerce/woocommerce-ios/commit/82dfbb86c6d893a8526ce449998d7b438dc5c5c4